spaceduck/src/Views/RegisterView.jsx

```diff
@@ -51,37 +92,66 @@ class Register extends React.Component {
                 'border': 'solid red 1px',
                 'background': 'rgba(255,0,0,0.3)'
             }
+             this.state.correctRepeatedPassword = false;
 
         }else{
-            inputStyle={}    
+            inputStyle={}   
+            this.state.correctRepeatedPassword = true; 
+        }
+        let passwordStyle = {};
+        // walidacja silnego hasła
+        if(!this.strongPassword(this.state.password)){
+            passwordStyle = {
+                'border': 'solid red 1px',
+                'background': 'rgba(255,0,0,0.3)'
+            }
+             this.state.correctPassword = false;
+        }else{
+            passwordStyle={};
+            this.state.correctPassword = true;
         }
 
-        //email validation
+        //walidacja emaila
         let emailStyle ={};
         if(!this.emailValidation(this.state.mail)){
             emailStyle={
                 'border': 'solid red 1px',
                 'background': 'rgba(255,0,0,0.3)'
             }
+             this.state.correctEmail = false;
         }else{
-            emailStyle ={};
+            emailStyle={};
+            this.state.correctEmail =true
+
+        }
+        //jeśli wszystkie dane są poprawne zmieniamy flagę do odblokowania przycisku
+        if(this.state.correctEmail && this.state.correctPassword && this.state.correctRepeatedPassword&& this.state.name!=''){
+            this.state.correctData=true;
+        }else{
+            this.state.correctData=false;
+        }
+       
+        //przenosimy po zarejestrowaniu na stronę logowania
```

Comment:
A z tymi komentarzami, to faktycznie dobrą praktyką jest jednak mieć je po angielsku

Reply:
Okiii będą po angielsku